src/appleseed/renderer/kernel/intersection/embreescene.h

```diff
@@ -42,6 +41,9 @@
 #include "foundation/utility/poolallocator.h"
 #include "foundation/utility/uid.h"
 
+// Embree headers.
+#include <embree3/rtcore.h>
```

Comment:
If this move is made, no need in EmbreeDevice object at all. It was introduce to hide embree guts from outer code.

Reply:
EmbreeScene is not part of the public API and it is not included by any public header.
Removing the private implementation saves a memory allocation and one indirection when accessing the class members.

I still think that EmbreeDevice can be useful as a wrapper, if we want to set options in the embree device, so I didn't remove it.

Reply:
Sounds reasonable, nice!